.github/workflows/go-test.yml

```diff
@@ -25,22 +25,33 @@ jobs:
         uses: actions/setup-go@fac708d6674e30b6ba41289acaab6d4b75aa0753 # v4.0.1
         with:
           go-version: ${{ matrix.go-version }}
-      
+
       - name: Checkout code
         uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
 
       - name: Create test directory
         run: |
           mkdir -p ${{ env.TEST_RESULTS_PATH }}/packer-plugin-sdk
-  
+
       - name: Run gofmt
         run: |
             make fmt-check
 
+      # Required for passing tests as we need to have a replace in the go.mod
+      # otherwise we'll get compile-time errors when running them
+      - name: Fix gocty
+        run: |
+          make install-gen-deps
+          packer-sdc fix .
+          go mod tidy
+          git config user.name "Packer CI"
+          git config user.email "ci@packer.io"
+          git add go.mod go.sum && git commit -m 'tmp changes for testing' # we need to add them as a commit to avoid `generate-check` reporting diffs
```

Comment:
Seeing as you are adding this step here to avoid the generation check what do you think about not doing this so early and just making it part of the test only. Preferably as part of the make test target so that locally we run packer-sdc fix as well for testing.

Does the generation change if the replace is in place?